From d2a57492af58fb56c594451bfbf5b62e8dfb4fb3 Mon Sep 17 00:00:00 2001
From: Christian Ehrlicher <ch.ehrlicher@gmx.de>
Date: Sun, 28 Sep 2025 17:33:03 +0200
Subject: [PATCH 245/553] Windows11Style: don't try to draw an icon when the
 rect is not valid

When the icon rect for CE_ItemViewItem is not valid we don't need to try
to draw the icon and also can skip some calculations.

Pick-to: 6.9
Change-Id: Ic5e448750712733196d21d5d56cf5cb4f1d0a106
Reviewed-by: Wladimir Leuschner <wladimir.leuschner@qt.io>
(cherry picked from commit e49ad95bb6e668aabf25f2bec3ab58afd7379bca)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 .../styles/modernwindows/qwindows11style.cpp     | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/src/plugins/styles/modernwindows/qwindows11style.cpp b/src/plugins/styles/modernwindows/qwindows11style.cpp
index e0a1c0391c2..a4c997f1203 100644
--- a/src/plugins/styles/modernwindows/qwindows11style.cpp
+++ b/src/plugins/styles/modernwindows/qwindows11style.cpp
@@ -1785,13 +1785,15 @@ void QWindows11Style::drawControl(ControlElement element, const QStyleOption *op
             }
 
             // draw the icon
-            QIcon::Mode mode = QIcon::Normal;
-            if (!(vopt->state & QStyle::State_Enabled))
-                mode = QIcon::Disabled;
-            else if (vopt->state & QStyle::State_Selected)
-                mode = QIcon::Selected;
-            QIcon::State state = vopt->state & QStyle::State_Open ? QIcon::On : QIcon::Off;
-            vopt->icon.paint(painter, iconRect, vopt->decorationAlignment, mode, state);
+            if (iconRect.isValid()) {
+                QIcon::Mode mode = QIcon::Normal;
+                if (!(vopt->state & QStyle::State_Enabled))
+                    mode = QIcon::Disabled;
+                else if (vopt->state & QStyle::State_Selected)
+                    mode = QIcon::Selected;
+                QIcon::State state = vopt->state & QStyle::State_Open ? QIcon::On : QIcon::Off;
+                vopt->icon.paint(painter, iconRect, vopt->decorationAlignment, mode, state);
+            }
 
             painter->setPen(highlightCurrent && highContrastTheme ? vopt->palette.base().color()
                                                                   : vopt->palette.text().color());
-- 
2.51.2

